controller/attribute/util.py

```diff
@@ -81,16 +83,18 @@ def run_attribute_calculation_exec_env(
         network=exec_env_network,
     )
 
-    logs = [
-        line.decode("utf-8").strip("\n")
-        for line in container.logs(
-            stream=True, stdout=True, stderr=True, timestamps=True
-        )
-    ]
+    logs = []
+    __update_progress(project_id=project_id, attribute_id=attribute_id, logs=logs, progress=0.0)
+    for line in container.logs(stream=True, stdout=True, stderr=True, timestamps=True):
+        line = line.decode("utf-8").strip("\n")
+        logs.append(line)
 
-    attribute.update(
-        project_id=project_id, attribute_id=attribute_id, logs=logs, with_commit=True
-    )
+        splitted_line = line.split(":")
+        if len(splitted_line) > 1:
+            if splitted_line[-2] == "progress":
+                __update_progress(project_id=project_id, attribute_id=attribute_id, logs=logs, progress=round(float(splitted_line[-1]),2))
+        
+    __update_progress(project_id=project_id, attribute_id=attribute_id, logs=logs, progress=1.0)
```

Comment:
does this work? for lf exec env I needed to run a thread to access the logs (read_container_logs_thread) since collecting the logs does it only after the container is completed which would result in the progress jumping from 0 to 100?

Reply:
will refactor it with a threaded solution